scipy/optimize/_zeros_py.py

```diff
@@ -1392,3 +1392,125 @@ def toms748(f, a, b, args=(), k=1,
                           maxiter=maxiter, disp=disp)
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag))
+
+
+def _bracketer(f, a, b=None, min=None, max=None, factor=2, batch=100):
+    """Bracket the root of a monotonic scalar function of one variable
+
+    Parameters
+    ----------
+    f : callable
+        The function for which the root is to be bracketed.
+    a, b : float
+        Starting guess of bracket, which need not contain a root. If `b` is
+        not provided, ``b = a + 1``.
+    min, max : float
+        Minimum and maximum allowable endpoints of the bracket, inclusive.
+    factor : float
+        The factor used to grow the bracket. See notes for details.
+    batch : int
+        The maximum number of arguments passed to the callable `f` at once.
```

Comment:
Maybe the default should be 1 in case `func` cannot accept being called in a vectorised fashion.

Reply:
Since this is private, I thought the default should be some number we think will tend to produce the best performance; it would be the developer's responsibility to set `batch=1` if the function they are solving might not allow array input.